classes/category/Repository.php

```diff
@@ -143,6 +147,58 @@ public function validate(?Category $object, array $props, array $allowedLocales,
 
         $errors = [];
 
+        if (isset($props['contextId'])) {
+            $validator->after(function (Validator $validator) use ($props, $context) {
+                if (!Application::getContextDAO()->exists($props['contextId'])) {
+                    $validator->errors()->add('contextId', __('api.contexts.404.contextNotFound'));
+                }
+                if ($context->getId() !== $props['contextId']) {
+                    $validator->errors()->add('contextId', __('api.categories.400.invalidContext'));
+                }
+            });
+        }
+
+        if (isset($props['path'])) {
+            $validator->after(function (Validator $validator) use ($props, $context, $object) {
+                // Check if path matches the allowed pattern
+                if (!preg_match(\Category::$PATH_REGEX, $props['path'] ?: '')) {
+                    $validator->errors()->add('path', __('grid.category.pathAlphaNumeric'));
+                }
+
+                $existingCategoryWithPath = Repo::category()->getCollector()
+                    ->filterByContextIds([$props['contextId']])
+                    ->filterByPaths([$props['path']])
+                    ->getMany()
+                    ->first();
+
+                $id = key_exists($props['categoryId'], $props) ? $props['categoryId'] : $object?->getId() ?? null;
+                $existingCategoryWithId = $id ? Repo::category()->get($id, $context->getId()) : null;
+
+                if ($existingCategoryWithPath && $existingCategoryWithPath->getPath() !== $existingCategoryWithId?->getPath()) {
+                    $validator->errors()->add('path', __('grid.category.pathExists'));
+                }
+            });
+        }
+
+        if (isset($props['image']) && $props['image']['temporaryFileId']) {
+            $validator->after(function (Validator $validator) use ($props, $context) {
+                $temporaryFileId = $props['image']['temporaryFileId'];
+                $temporaryFileManager = new TemporaryFileManager();
+                $user = Application::get()->getRequest()->getUser();
+                $temporaryFile = $temporaryFileManager->getFile((int)$temporaryFileId, $user->getId());
+                $imageExtension = $temporaryFile ? $temporaryFileManager->getImageExtension($temporaryFile->getFileType()) : [];
+                $isValidExtension = in_array($imageExtension, ['.jpg', '.png', '.gif']);
```

Comment:
If we're using this list of extensions somewhere else, I think it should be turned into a constant (e.g. `SUPPORTED_IMAGE_TYPES`).